File: rust/src/models/slot_plan_fee.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::term_rate::TermRate;

/// Slot-based fee (e.g., per-seat pricing)
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct SlotPlanFee {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub minimum_count: Option<i32>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub quota: Option<i32>,

    pub rates: Vec<TermRate>,

    pub slot_unit_name: String,
}

impl SlotPlanFee {
    pub fn new(rates: Vec<TermRate>, slot_unit_name: String) -> Self {
        Self {
            minimum_count: None,
            quota: None,
            rates,
            slot_unit_name,
        }
    }
}

File: typescript/src/models/slotPlanFee.ts
// this file is @generated
import { type TermRate, TermRateSerializer } from "./termRate";
/** Slot-based fee (e.g., per-seat pricing) */
export interface SlotPlanFee {
  minimumCount?: number | null;

  quota?: number | null;

  rates: TermRate[];

  slotUnitName: string;
}

export const SlotPlanFeeSerializer = {
  _fromJsonObject(object: any): SlotPlanFee {
    return {
      minimumCount: object["minimum_count"],
      quota: object["quota"],
      rates: object["rates"].map((item: any) => TermRateSerializer._fromJsonObject(item)),
      slotUnitName: object["slot_unit_name"],
    };
  },

  _toJsonObject(self: SlotPlanFee): any {
    return {
      minimum_count: self.minimumCount,
      quota: self.quota,
      rates: self.rates.map((item: any) => TermRateSerializer._toJsonObject(item)),
      slot_unit_name: self.slotUnitName,
    };
  },
};
